lib/travis/build/addons/apt_packages.rb

```diff
@@ -0,0 +1,43 @@
+require 'travis/build/addons/base'
+
+module Travis
+  module Build
+    class Addons
+      class AptPackages < Base
+        SUPER_USER_SAFE = true
+
+        def after_prepare
+          sh.fold 'apt_packages' do
+            sh.echo "Installing APT Packages (BETA)", ansi: :yellow
+
+            whitelisted = []
+            config.each do |package|
+              if whitelist.include?(package)
+                whitelisted << package
+              else
+                sh.echo "Ignoring unknown/disallowed package #{package.inspect}"
+              end
+            end
+
+            unless whitelisted.empty?
+              sh.export 'DEBIAN_FRONTEND', 'noninteractive', echo: true
+              sh.cmd "sudo -E apt-get -yq update", echo: true, timing: true
+              sh.cmd 'sudo -E apt-get -yq --no-install-suggests --no-install-recommends ' \
+                     "install #{whitelisted.join(' ')}", echo: true, timing: true
+            end
+          end
+        end
+
+        private
+
+          def config
+            Array(super)
+          end
+
+          def whitelist
+            @whitelist ||= ENV['TRAVIS_BUILD_APT_WHITELIST'].to_s.split(/\s*,\s*/)
```

Comment:
@BanzaiMan I thought of something, actually.  Having spaces in env vars means we always have to be careful with quoting, which is minor but definitely a pain.  For example:

``` bash
# hk env -a heroku-app-name
TRAVIS_BUILD_APT_WHITELIST=foo bar ham bones
```

... meaning I will always have to munge this output a la `s/=/="/;s/$/"/`.  Again, it's minor, but a pain :scream_cat:

Would you accept `:`-delimited?